radio/src/opentx.h

```diff
@@ -1182,13 +1182,13 @@ extern ReusableBuffer reusableBuffer;
 uint8_t zlen(const char *str, uint8_t size);
 bool zexist(const char *str, uint8_t size);
 unsigned int effectiveLen(const char * str, unsigned int size);
-char * strcat_zchar(char *dest, const char *name, uint8_t size, const char *defaultName=nullptr, uint8_t defaultNameSize=0, uint8_t defaultIdx=0);
-#define strcatFlightmodeName(dest, idx) strcat_zchar(dest, g_model.flightModeData[idx].name, LEN_FLIGHT_MODE_NAME, STR_FM, PSIZE(TR_FM), idx+1)
+char * strcat_zchar(char *dest, const char *name, uint8_t size, const char spaceSym = 0, const char *defaultName=nullptr, uint8_t defaultNameSize=0, uint8_t defaultIdx=0);
+#define strcatFlightmodeName(dest, idx) strcat_zchar(dest, g_model.flightModeData[idx].name, LEN_FLIGHT_MODE_NAME, 0, STR_FM, PSIZE(TR_FM), idx+1)
 #if defined(EEPROM)
-#define strcat_modelname(dest, idx) strcat_zchar(dest, modelHeaders[idx].name, LEN_MODEL_NAME, STR_MODEL, PSIZE(TR_MODEL), idx+1)
-#define strcat_currentmodelname(dest) strcat_modelname(dest, g_eeGeneral.currModel)
+#define strcat_modelname(dest, idx) strcat_zchar(dest, modelHeaders[idx].name, LEN_MODEL_NAME, 0, STR_MODEL, PSIZE(TR_MODEL), idx+1)
```

Comment:
Changing `0` to `0x20` seems to be the last part of the puzzle for B&W... I'm not sure if this is the right change, or if it breaks something elsewhere though?
```
#define strcat_modelname(dest, idx) strcat_zchar(dest, modelHeaders[idx].name, LEN_MODEL_NAME, 0x20, STR_MODEL, PSIZE(TR_MODEL), idx+1)
```

Reply:
Well, 0x20 IS an ASCII for "whitespsce", isn't it? 
I'm surprised compiler has a different opinion here, but, most likely, the combination of symbols ' ' is parsed incorrectly.

It is difficult to judge what can be broken: the OTX code for b/w radios has a long history. E.g. it turned out that different code is used to display model notes when loading the model during the boot and  when switching to another model.

If there is no way to rebase these legacy GUIs to current OTX code, we may have difficult time with re-implementing or porting this stuff manually.
It may be especially difficult if we want new UI features supported on the old platforms.

Reply:
Welcome to the OTX devs' world